From b39713e78a4a9c8828e7876bbd21c4a1973d28d6 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Fri, 22 May 2015 13:24:27 -0700
Subject: [PATCH 085/441] sdhci: Cleanup *_EMMC_HW_MUTEX macros

In LOCK_EMMC_HW_MUTEX and UNLOCK_EMMC_HW_MUTEX macros, move
(un)lock_emmc_access() and (en|dis)able_irq() calls into blocks
protected by netss_driver_ready();
---
 include/linux/mmc/sdhci.h | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

--- a/include/linux/mmc/sdhci.h
+++ b/include/linux/mmc/sdhci.h
@@ -219,19 +219,23 @@ static void unlock_emmc_access(void)
   if (((struct sdhci_host *)host->private)->flags & SDHCI_SUPPORT_HW_MUTEX)\
     {\
       if (netss_driver_ready()) \
+      {\
         hw_mutex_lock(HW_MUTEX_EMMC);\
-      lock_emmc_access();\
-      enable_irq(((struct sdhci_host *)host->private)->irq);\
+        lock_emmc_access();\
+        enable_irq(((struct sdhci_host *)host->private)->irq);\
+      }\
     }\
   } while(0)
 
 #define UNLOCK_EMMC_HW_MUTEX(host) do{\
   if (((struct sdhci_host *)host->private)->flags & SDHCI_SUPPORT_HW_MUTEX)\
     {\
-      disable_irq(((struct sdhci_host *)host->private)->irq);\
-      unlock_emmc_access();\
       if (netss_driver_ready()) \
+      {\
+        disable_irq(((struct sdhci_host *)host->private)->irq);\
+        unlock_emmc_access();\
         hw_mutex_unlock(HW_MUTEX_EMMC);\
+      }\
     }\
   } while(0)
 
